src/vs/workbench/common/editor.ts

```diff
@@ -789,6 +789,7 @@ export interface IEditorIdentifier {
 
 export interface IEditorContext extends IEditorIdentifier {
 	event?: any;
+	resource?: URI;
```

Comment:
@isidorn would it not be enough to call `editorinput.getResource()`? Every editor now has this method so you can get the resource from there.

Reply:
@bpasero notice that in some actions I only have a `resource` and then I have to pass them on to the command somehow without having the editor.
Currently I do 
`return this.commandService.executeCommand(OPEN_TO_SIDE_COMMAND_ID, { resource: this.resource });`

An ugly alternative is 
`
return this.commandService.executeCommand(OPEN_TO_SIDE_COMMAND_ID, { editor: { getResource: () => this.resource }});
`